src/noisepy/seis/S1_fft_cc_MPI.py

```diff
@@ -47,350 +48,100 @@
-            for itag in range(len(all_tags)):
-                if flag:
-                    print("working on station %s and trace %s" % (sta, all_tags[itag]))
-
-                # read waveform data
-                if input_fmt == "h5":
-                    source = ds.waveforms[tmps][all_tags[itag]]
-                else:
-                    source = obspy.read(tmps)
-                    inv1 = noise_module.stats2inv(source[0].stats, fc_para, locs)
-                    sta, net, lon, lat, elv, loc = noise_module.sta_info_from_inv(inv1)
-
-                # channel info
-                comp = source[0].stats.channel
-                if comp[-1] == "U":
-                    comp.replace("U", "Z")
-                if len(source) == 0:
-                    continue
-
-                # cut daily-long data into smaller segments (dataS always in 2D)
-                trace_stdS, dataS_t, dataS = noise_module.cut_trace_make_stat(
-                    fc_para, source
-                )  # optimized version:3-4 times faster
-                if not len(dataS):
-                    continue
-                N = dataS.shape[0]
-
-                # do normalization if needed
-                source_white = noise_module.noise_processing(fc_para, dataS)
-                Nfft = source_white.shape[1]
-                Nfft2 = Nfft // 2
-                if flag:
-                    print("N and Nfft are %d (proposed %d),%d (proposed %d)" % (N, nseg_chunk, Nfft, nnfft))
-
-                # keep track of station info to write into parameter section of ASDF files
-                station.append(sta)
-                network.append(net)
-                channel.append(comp)
-                clon.append(lon)
-                clat.append(lat)
-                location.append(loc)
-                elevation.append(elv)
-
-                # load fft data in memory for cross-correlations
-                data = source_white[:, :Nfft2]
-                fft_array[iii] = data.reshape(data.size)
-                fft_std[iii] = trace_stdS
-                fft_flag[iii] = 1
-                fft_time[iii] = dataS_t
-                iii += 1
-                del trace_stdS, dataS_t, dataS, source_white, data
-
-        if input_fmt == "h5":
-            del ds
+        fft_array = np.zeros((nchannels, nseg_chunk * (nnfft // 2)), dtype=np.complex64)
+        fft_std = np.zeros((nchannels, nseg_chunk), dtype=np.float32)
+        fft_flag = np.zeros(nchannels, dtype=np.int16)
+        fft_time = np.zeros((nchannels, nseg_chunk), dtype=np.float64)
+
+        logger.debug(f"nseg_chunk: {nseg_chunk}, nnfft: {nnfft}")
+        # loop through all channels
+        N: int
+        Nfft2: int
+        for ix_ch, ch in enumerate(channels):
+            ch_data = raw_store.read_data(ts, ch)
+            # TODO: Below the last values for N and Nfft are used?
+            fft_array[ix_ch], fft_std[ix_ch], fft_time[ix_ch], N, Nfft2 = compute_fft(fft_params, ch_data)
```

Comment:
I found it a bit strange that `N` is defined within the loop and then the last value that N got in this loop is what ends up being used for the next steps. But this is the way it was already working (see line 337 in the red above: `N = dataS.shape[0]`). Is this ok because N is expected to be the same across all channels?